snapcraft/plugins/go.py

```diff
@@ -111,7 +112,7 @@ def pull(self):
         super().pull()
         os.makedirs(self._gopath_src, exist_ok=True)
 
-        if self.options.source:
+        if glob('{}/**/*.go'.format(self.sourcedir), recursive=True):
```

Comment:
Although... wouldn't having _anything_ in `self.sourcedir` mean that `source` was set? Why not just check to see if `self.sourcedir` is empty?